src/main/webapp/definitions/alfresco-core.yaml

```diff
@@ -2633,6 +2641,126 @@ paths:
+        
+        **Note:** The download node can be deleted using the **DELETE /nodes/{downloadId}** endpoint
+        
+        By default, if the download node is not deleted it will be picked up by a cleaner job which removes download nodes older than a configurable amount of time(default is 1 hour)
+        
+      operationId: createDownload
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/fieldsParam'
+        - in: body
+          name: DownloadBodyCreate
+          description: The nodeIds the content of which will be zipped, which zip will be set as the content of our download node.
+          required: true
+          schema:
+            $ref: '#/definitions/DownloadBodyCreate'
+      responses:
+        '202':
+          description: The request to create a download node was accepted
+          schema:
+            $ref: '#/definitions/DownloadEntry'
+        '400':
+          description: |
+            Invalid parameter: **DownloadBodyCreate** is invalid due to incorrect syntax, or **nodeIds** is empty, or **nodeIds** contains a duplicate **nodeId**
+        '401':
+          description: Authentication failed
+        '403':
+          description: Current user does not have permission to add a certain **nodeId** to the zip
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+            
+  '/downloads/{downloadId}':
+    get:
+      x-alfresco-since: "5.2.1"
+      tags:
+        - downloads
+      summary: Get a download
+      description: |
+        **Note:** this endpoint is available in Alfresco 5.2.1 and newer versions.
+        
+        Retrieve status information for download node **downloadId**
+        
+      operationId: getDownload
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/downloadIdParam'
+        - $ref: '#/parameters/fieldsParam'
+      responses:
+        '200':
+          description: Download node information
+          schema:
+            $ref: '#/definitions/DownloadEntry'
+        '400':
+          description: |
+            Invalid parameter: **downloadId** is invalid, or **downloadId** does not point to a node of download type
+        '401':
+          description: Authentication failed
+        '403':
+          description: Current user does not have permission to get information about **downloadId** node
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    delete:
+      x-alfresco-since: "5.2.1"
+      tags:
+        - downloads
+      summary: Cancel a download
+      description: |
+        **Note:** this endpoint is available in Alfresco 5.2.1 and newer versions.
+        
+        This signals the download node that the zip is not needed anymore. Thus any archiving going on is stopped in its tracks. 
```

Comment:
Hi Gavin. 
We agreed to have the DELETE do both(cancel and delete the node) in the light of the fact that a non admin user would not be able to delete a node of type download:download.

But it seems that as long as an user is the owner he can delete the node.

Hence in order to prevent any confusion, after a discussion with Jamal, we considered that it's better for the DELETE /downloads/id to do just one thing, namely cancel the download.
Should the user decide that the created archive is too big to just lie around to be picked up by the cleaner job(which runs every hour by default), he could use the DELETE /nodes/id to delete the node.

Moreover, since there isn't a GET /downloads/id/content api to download the archive, there shouldn't be a DELETE /downloads/id to delete the node. Instead, the existing APIs should be used, GET /nodes/id/content, DELETE /nodes/id

Thanks. I'll update the description to be clearer and more formal.